api/app/signals/apps/email_integrations/signal_receivers.py

```diff
@@ -3,9 +3,40 @@
 from django.dispatch import receiver
 
 from signals.apps.email_integrations import tasks
-from signals.apps.signals.managers import create_initial, update_status
+from signals.apps.signals.managers import (
+    create_initial,
+    update_signal_departments,
+    update_status,
+    update_user_assignment
+)
 
 
 @receiver((create_initial, update_status), dispatch_uid='core_email_integrations_update_status')
 def update_status_handler(sender, signal_obj, *args, **kwargs):
     tasks.send_mail_reporter.delay(pk=signal_obj.pk)
+
+
+@receiver(update_signal_departments, dispatch_uid='core_email_integrations_update_signal_departments')
+def update_signal_departments_handler(sender, signal_obj, signal_departments, prev_signal_departments, *args, **kwargs):
+    if signal_departments and prev_signal_departments:
+        if set(signal_departments.departments.all()) == set(prev_signal_departments.departments.all()):
+            return  # do not trigger when the departments field is unchanged
+
+    department_pks = [department.pk for department in signal_departments.departments.all()]
+
+    tasks.send_mail_assigned_signal_departments.delay(
+        signal_pk=signal_obj.pk,
+        department_pks=department_pks
+    )
+
+
+@receiver(update_user_assignment, dispatch_uid='core_email_integrations_update_user_assignment')
+def update_user_assignment_handler(sender, signal_obj, user_assignment, prev_user_assignment, *args, **kwargs):
+    if user_assignment and prev_user_assignment:
```

Comment:
Please move these specific checks if the email action can be triggered to a specific rule `AssignedRule`.

Reply:
@vanbuiten and I discussed this yesterday and decided to keep the business logic in tasks.py for now and use the different usages of MailService later as an inspiration for refactoring.